tests/test_dynamic_boundary_conditions/tide/test_tide_data_from_niwa.py

```diff
@@ -0,0 +1,82 @@
+import unittest
+from unittest.mock import patch
+from typing import List, Dict
+
+from shapely import LineString
+from datetime import date, timedelta
+from shapely.wkt import loads
+import pandas as pd
+import geopandas as gpd
+
+import numpy as np
+
+from src.dynamic_boundary_conditions.tide import tide_data_from_niwa
+
+
+class TideDataFromNiwaTest(unittest.TestCase):
+
+    query_loc_row = None
+
+    @classmethod
+    def setUpClass(cls):
+        data_dir = "tests/test_dynamic_boundary_conditions/tide/data"
+        cls.query_loc_row = pd.read_csv(f"{data_dir}/query_loc_row.txt")
+        cls.query_loc_row['geometry'] = cls.query_loc_row['geometry'].apply(loads)
+        cls.gdf = gpd.GeoDataFrame(cls.query_loc_row, geometry='geometry')
+
+    def test_get_query_loc_coords_position(self):
+        actual_output = tide_data_from_niwa.get_query_loc_coords_position(self.gdf)
+        expected_output = (-43.39127633722616, 172.7098878483352, 'right')
+        self.assertEqual(actual_output, expected_output)
+
+    def test_valid_input_get_date_ranges(self):
+        # Test with default values
+        result = tide_data_from_niwa.get_date_ranges()
+        expected_result = {date.today(): 31}
+        self.assertEqual(result, expected_result)
```

Comment:
Generally best practice to only test for one result per test method. So split up default values and custom values